File: PiBlynk-py/05-rlogger.py


import socket

from PiBlynk import Blynk
from mytoken import *
blynk = Blynk(token)


#------------------------------------------
import time
def timeNow():
    return time.asctime()[11:19]

#------------------------------------


# rlogger

rlctr = 0   
def timer3loop(s):  
    # write to (terminal style) rlogger at APP on vp 18
    global rlctr
    rlctr = rlctr+1
    blynk.virtual_write(18, timeNow()+ " "+str(rlctr) + "\n"); 

blynk.add_Task( 6, timer3loop)   # for every 6 secs


def cnct_cb():
    print ("Connected: "+ timeNow())
    blynk.virtual_write(18, "\n"+socket.gethostname()+"\n")
blynk.on_connect(cnct_cb)


######################################################################################

blynk.run()

######################################################################################


# At APP:
# make terminal widget on that V18, fill whole phone screen, 
# turn off its input box
# now it is a live-time remote log display screen rpi->app  with rlogger.write()
# but server WONT indefinitely buffer messages if APP is offline.

File: piblynk-js/05-rlogger.js
#!/usr/bin/env nodejs


var print = console.log;   // create an alias to look like python print() !!!

require("./mytoken"); // fetch auth token from sep file
var blynk = new (require("piblynk")).Blynk(token);  
blynk.on('connect', function() { print("Blynk up. " + timeNow());  });


//-------------------------

function timeNow() {
  var d = new Date();
  var min  = d.getMinutes();
  min = (min < 10 ? "0" : "") + min;
  var sec  = d.getSeconds();
  sec = (sec < 10 ? "0" : "") + sec;
  var hrs  = d.getHours();
  hrs = (hrs < 10 ? "0" : "") + hrs;
  return(hrs + ":" + min + ":" + sec);
};


//-----------------------------------

myIP = require('my-local-ip')();
myPID = process.pid;
myHostname = require("os").hostname();




ctr=0;
rlogger = new blynk.VirtualPin(18);

// initial message out:
setTimeout(function(){
	rlogger.write("\nip:"+myIP+"  pid:"+myPID+"  "+myHostname+"\n");
	}, 2000);  // slight delay: allowing for blynk to get connected to server

 
// dummy regular text to logger  
setInterval(function(){ 
  rlogger.write(timeNow()+ " "+ctr++ + "\n"); 
   print(timeNow());

  },5000);

/*
 * At APP:
 * make terminal widget on that V18, fill whole phone screen, 
 * turn off its input box
 * now it is a live-time remote log display screen rpi->app  with rlogger.write()
 * but server WONT indefinitely buffer messages if APP is offline.
*/



